### ISSUE TITLE
Feature: Do not finish completed anima quest if no bag space

### ISSUE THREAD

**Issue by neocryptek** [TLDRMissions/TLDRMissions#620]
**TLDR Version of your idea**
Don't finish completed anima missions if character has no bag space.

**Slightly longer version**
If a character finishes an anima mission, but has no bag space, the anima is wasted. Anima goes into the mail, but does not count towards the weekly anima quest when retrieved from mail.

**Additional context**
If a player only logs into a character to send missions, their bag space can fill up over time throughout the week. A player may not be able to empty bag space at the moment, but still wants to run missions safely. The anima mission block setting tries to help the player safely run anima missions, without having to think about edge cases where anima is wasted. Anima earned without bag space is wasted.

Small tweak to logic added in #613 and #616 to check if character has available bag space for the anima rewards to go into. Should be run before each mission is finished, to catch if character runs out of bag space in the middle of finishing multiple missions (ex: 5 anima missions done, but only 3 bag spots left: If finish all 5 the last 2 will not fit). Should not consider reagent/etc special bag space as anima cannot go into those bags.

An easy way to fill up your bag space for testing is to acquire a stackable item (ex: 200 cloth), then split the stack into singles to fill up every bag slot. Clicking the clean up bags feature will re-stack the items opening the bag slots back up again quickly.

Thank you! This feature has saved me a ton of time as I've been doing the same manually for a while now.

**Comment by neocryptek** [comment]
Hmm, I tried this out today and it didn't seem to work. I had 1 bag slot open, made sure the setting was on, 3 anima missions ready, clicked finish button. All 3 anima missions were finished, one went into my bags and the other two into my mail: 2 wasted. No chat "anima blocked" messages like the other settings, so it probably didn't trigger.

I have a reagent bag, but it was also full at the time. So not sure where it thought I had bag space at?

Thanks!

**Comment by teelolws** [comment]
That will be because I only put the bag check at the start, not in between missions too.

**Comment by neocryptek** [comment]
Is `areBagsFull()` running right before each anima mission is finished, or only once at the start when the button is clicked?

> Should be run before each mission is finished, to catch if character runs out of bag space in the middle of finishing multiple missions (ex: 5 anima missions done, but only 3 bag spots left: If finish all 5 the last 2 will not fit).

**Comment by teelolws** [comment]
Just tested it with the bag check in between missions but it didn't work. Studying it now.

**Comment by teelolws** [comment]
Because there is a delay between finishing the mission, and the item going into your bags, I won't be able to do accurate bag checks and still maintain the speed of finishing the missions.

I'd rather not get into the realm of checking stackable item limits and such so I will go with a more... false-positive prone approach. If you don't have enough bag slots for all the anima missions, it will block them all. Even if some of those anima rewards could have gone in to existing stacks.

**Comment by neocryptek** [comment]
Maybe it runs the check too fast vs item going into bags?

Edit: beat me again :)
